inductiva/resources/machines_base.py

```diff
@@ -68,13 +73,26 @@ def __init__(
         if data_disk_gb <= 0:
             raise ValueError("`data_disk_gb` must be positive.")
 
+        if auto_resize_disk_max_gb is not None:
+            if not isinstance(auto_resize_disk_max_gb,
+                              int) or auto_resize_disk_max_gb <= 0:
+                raise ValueError(
+                    "`auto_resize_disk_max_gb` must be a positive integer.")
+
+            if auto_resize_disk_max_gb < data_disk_gb:
```

Comment:
is this strict enough? potentially we could ask the auto_resize_disk_max_gb to be at least size_increment_gb bigger than data_disk_gb... Not sure what happens if you set the value to auto_resize_disk_max_gb = data_disk_gb -- @luispcunha ?

Also, maybe we should suggest that it should be at least 2x bigger than the data_disk_gb, otherwise it's not worth using this mechanism?

Reply:
@luispcunha does the max size need to be a factor of the increment? 
does this work?
data_disk_gb - 10
max_disk_size_gb - 20
increment - 3

Reply:
I've changed this check to
```
if auto_resize_disk_max_gb < data_disk_gb + self._size_increment_gb:
                raise ValueError("`auto_resize_disk_max_gb` must be greater "
                                 "than or equal to `data_disk_gb + "
                                 f"{self._size_increment_gb}GB`.")
```